File: go-sdk/pkg/models/operations/gettransaction.go
package operations

import (
	"github.com/speakeasy-sdks/formance-sdks/go-client-sdk/pkg/models/shared"
)

type GetTransactionPathParams struct {
	Ledger string `pathParam:"style=simple,explode=false,name=ledger"`
	Txid   int64  `pathParam:"style=simple,explode=false,name=txid"`
}

type GetTransactionRequest struct {
	PathParams GetTransactionPathParams
}

type GetTransactionResponse struct {
	ContentType         string
	ErrorResponse       *shared.ErrorResponse
	StatusCode          int64
	TransactionResponse *shared.TransactionResponse
}

File: ts-sdk/src/sdk/models/operations/gettransaction.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetTransactionPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ledger" })
  ledger: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=txid" })
  txid: number;
}

export class GetTransactionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetTransactionPathParams;
}

export class GetTransactionResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorResponse?: shared.ErrorResponse;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  transactionResponse?: shared.TransactionResponse;
}